zebra/zebra_nhg.c

```diff
@@ -1071,25 +1072,35 @@ static void zebra_nhg_set_valid(struct nhg_hash_entry *nhe, bool valid)
 
 	/* Update validity of nexthops depending on it */
 	frr_each (nhg_connected_tree, &nhe->nhg_dependents, rb_node_dep) {
+		dependent_valid = valid;
 		if (!valid) {
 			/*
 			 * Grab the first nexthop from the depending nexthop group
 			 * then let's find the nexthop in that group that matches
 			 * my individual nexthop and mark it as no longer ACTIVE
 			 */
 			struct nexthop *nexthop = rb_node_dep->nhe->nhg.nexthop;
+			struct nexthop *nexthop_invalid = NULL;
 
 			while (nexthop) {
 				if (nexthop_same(nexthop, nhe->nhg.nexthop))
-					break;
-
+					nexthop_invalid = nexthop;
```

Comment:
Shouldn't we do UNSET_FLAG here instead of outside the loop?

Reply:
Fixed this